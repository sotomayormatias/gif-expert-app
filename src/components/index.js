export * from './gifGrid';
export * from './gifItem';
export * from './input';