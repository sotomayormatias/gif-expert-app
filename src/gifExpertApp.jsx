import React from "react";
import { useState } from "react";
import { GifGrid } from "./components/gifGrid";
import { Input } from "./components/input";

export const GifExpertApp = () => {

    const [ categories, setCategories ] = useState([]);

    const handleSearch = (input) => {
        setCategories([input, ...categories])
    }

    return (
        <>
            <h1>Gif Expert App</h1>

            <Input onSearch={handleSearch} />
            {categories.map((category)=>(
                <GifGrid key={category} category={category} />
            ))}
        </>
    )
};
