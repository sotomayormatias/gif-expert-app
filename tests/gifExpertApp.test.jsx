import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/gifExpertApp"
import React from "react";

describe('Test on <GifExpertApp />', () => {
    test('Should match snapshot', () => {
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('Should add categories on new search', () => {
        const inputValue = 'boquita';
        const inputValue2 = 'boca juniors';
        const setState = jest.fn();
        jest
          .spyOn(React, 'useState')
          .mockImplementationOnce(initState => [initState, setState]);

        render(<GifExpertApp />);
        
        const input = screen.getByRole('textbox');
        fireEvent.input(input, {target: {value: inputValue}});
        fireEvent.input(input, {target: {value: inputValue2}});

        expect(setState).toHaveBeenCalledTimes(2);
    });
})