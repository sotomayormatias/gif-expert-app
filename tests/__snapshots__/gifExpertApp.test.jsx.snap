// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifExpertApp /> Should match snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar"
      type="text"
      value=""
    />
  </form>
</div>
`;
